clab/clab.go

```diff
@@ -168,45 +168,67 @@ func (c *CLab) GlobalRuntime() runtime.ContainerRuntime {
 	return c.Runtimes[c.globalRuntime]
 }
 
-func (c *CLab) CreateNodes(ctx context.Context, workers uint) {
+func (c *CLab) CreateNodes(ctx context.Context, maxWorkers uint, serialNodes map[string]struct{}) {
+
 	wg := new(sync.WaitGroup)
-	wg.Add(int(workers))
-	nodesChan := make(chan nodes.Node)
-	// start workers
-	for i := uint(0); i < workers; i++ {
-		go func(i uint) {
-			defer wg.Done()
-			for {
-				select {
-				case node, ok := <-nodesChan:
-					if node == nil || !ok {
-						log.Debugf("Worker %d terminating...", i)
-						return
-					}
-					log.Debugf("Worker %d received node: %+v", i, node.Config())
-					// PreDeploy
-					err := node.PreDeploy(c.Config.Name, c.Dir.LabCA, c.Dir.LabCARoot)
-					if err != nil {
-						log.Errorf("failed pre-deploy phase for node %q: %v", node.Config().ShortName, err)
-						continue
-					}
-					// Deploy
-					err = node.Deploy(ctx)
-					if err != nil {
-						log.Errorf("failed deploy phase for node %q: %v", node.Config().ShortName, err)
-						continue
-					}
-				case <-ctx.Done():
+	if len(serialNodes) > 0 {
+		wg.Add(int(maxWorkers) + 1)
+	} else {
+		wg.Add(int(maxWorkers))
+	}
```

Comment:
I would write this as:
```golang
wg.Add(int(maxWorkers))
if len(serialNodes) > 0 {
  wg.Add(1)
}
```

But I just don't like `else`

Reply:
did this